src/v2/api/services/PollService.ts

```
@@ -127,36 +126,55 @@ export class PollService {
           },
         },
       },
-      include: {
-        questionAnswers: {
-          where: {
-            disciplineTeacher: {
-              teacherId,
-              discipline: {
-                subjectId: query.subjectId,
-                year: query.year,
-                semester: query.semester,
-              },
-            },
+      include: undefined,
+    };
+    const questions = await this.questionRepository.findMany((questionsData));
+
+    const commentsData = {
+      where: {
+        disciplineTeacher: {
+          teacherId,
+          discipline: {
+            subjectId: query.subjectId,
```

You should throw an exception if `subjectId` is invalid